### PR TITLE
[WIP] add BoxDecoration back to dialog and action sheet

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
## Description

Add the `BoxDecoration` with overlay blendMode back, to mitigate https://github.com/flutter/flutter/issues/50183. Breaks one test.

## Screenshots
Looks fine to me in general but there's an unwanted glowing effect in dark mode around the action sheet.

Light Mode  | Dark Mode
 --- | --- 
Simulator Screen Shot - iPhone 11 Pro - 2020-02-25 at 10 08 07 | Simulator Screen Shot - iPhone 11 Pro - 2020-02-25 at 10 09 06 
Simulator Screen Shot - iPhone 11 Pro - 2020-02-25 at 10 08 32 | Simulator Screen Shot - iPhone 11 Pro - 2020-02-25 at 10 09 13
Simulator Screen Shot - iPhone 11 Pro - 2020-02-25 at 10 10 43 | Simulator Screen Shot - iPhone 11 Pro - 2020-02-25 at 10 11 07

## Related Issues

https://github.com/flutter/flutter/issues/50183

## Tests

I added the following tests:

- Dialog contains BoxDecoration with overlay blend mode.
- Action sheets contain BoxDecoration with overlay blend mode

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by flar** [comment]
I'm not sure I'm seeing the glowing that you are talking about around the dark mode action sheets. Where should I be looking?

File: packages/flutter/lib/src/cupertino/action_sheet.dart [hunk 1]
```diff
@@ -30,6 +30,20 @@ const TextStyle _kActionSheetContentStyle = TextStyle(
...
+// This decoration is applied to the blurred backdrop to lighten the blurred
+// image. Brightening is done to counteract the dark modal barrier that
+// appears behind the alert. The overlay blend mode does the brightening.
+// The white color doesn't paint any white, it's just the basis for the
+// overlay blend mode.
+//
+// The decoration is needed for `CupertinoActionSheet` to not appear excessively
+// transparent when placed above a PlatformView. See
+// https://github.com/flutter/flutter/issues/50183.
+const BoxDecoration _kAlertBlurOverlayDecoration = BoxDecoration(
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/action_sheet.dart:42]: We had this before there was dark mode. Now we may need 2 to handle the right visual fidelity.

File: packages/flutter/test/cupertino/dialog_test.dart [hunk 2]
```diff
@@ -91,7 +91,9 @@ void main() {
```
Comment (xster) [inline packages/flutter/test/cupertino/dialog_test.dart:91]: Can we add a golden test to check for the new bug specifically?

**Comment by LongCatIsLooong** [comment]
Sorry this has been sitting here for way too long now. I'll follow up on the reviews.

**Comment by xster** [comment]
@LongCatIsLooong do you want to refresh this PR?

**Comment by LongCatIsLooong** [comment]
@xster I need https://github.com/flutter/engine/pull/20309 to implement the visual effects. I'll continue working on this one when the color filter PR gets in.

**Comment by Hixie** [comment]
@LongCatIsLooong Since this is over a year old I'm going to close it to get it off the list but feel free to reopen when it's ready.

**Pull request closed without merging**